compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -388,28 +403,45 @@ void RecordMinMax::profileDataWithRandomInputs(const std::string &mode, float mi
       if (num_elements == 0)
         throw std::runtime_error("Only support non-zero sized inputs");
 
-      // TODO Support more input data types
-      assert(input_node->dtype() == loco::DataType::FLOAT32 ||
-             input_node->dtype() == loco::DataType::BOOL);
-
-      if (input_node->dtype() == DataType::FLOAT32)
+      switch (input_node->dtype())
```

Comment:
this and below lines dones' seem to be related with the title/explanation.
please put these changes to another PR.